heron/common/src/java/org/apache/heron/common/utils/metrics/FullSpoutMetrics.java

```diff
@@ -41,9 +42,12 @@
  * 2. New them in the constructor
  * 3. Register them in registerMetrics(...) by using MetricsCollector's registerMetric(...)
  * 4. Expose methods which could be called externally to change the value of metrics
+ *
+ * This is a spout metrics object with more information and it is used in heron core. To change to a faster
+ * metrics object, go to SpoutInstance.java and replace "new FullSpoutMetrcs" to "new SpoutMetrics"
  */
 
-public class FullSpoutMetrics extends SpoutMetrics {
+public class FullSpoutMetrics implements ISpoutMetrics {
```

Comment:
The original inheritance relationships is `FullSpoutMetrics extends SpoutMetrics implements ComponentMetrics`. But with the change here, the relationship chain is separate and now are `FullSpoutMetrics implements ISpoutMetrics` and `SpoutMetrics implements ISpoutMetrics`.

Many metrics originally available in `FullSpoutMetrics` which are inherited from `SpoutMetrics` are now missing.

Reply:
There is no inheritance between them.